modules/nickserv/info.c

```diff
@@ -198,40 +210,40 @@ ns_cmd_info(struct sourceinfo *si, int parc, char *parv[])
 			command_success_nodata(si, _("Real addr  : %s"), md->value);
 	}
 
-	if (recognized)
+	if (!hide_info && recognized)
 		command_success_nodata(si, _("Recognized : now (matches access list)"));
 
 	// show nick's lastseen/online, if we have a nick
-	if (u != NULL)
+	if (!hide_info && u != NULL)
 		command_success_nodata(si, _("Last seen  : now"));
 	else if (mn != NULL)
 	{
 		tm2 = localtime(&mn->lastseen);
 		strftime(lastlogin, sizeof lastlogin, TIME_FORMAT, tm2);
 		if (hide_info)
-			command_success_nodata(si, _("Last seen  : (about %u week(s) ago)"), (unsigned int)((CURRTIME - mn->lastseen) / SECONDS_PER_WEEK));
+			command_success_nodata(si, _("Last seen  : %s"), ns_obfuscate_time_ago(mn->lastseen));
 		else
 			command_success_nodata(si, _("Last seen  : %s (%s ago)"), lastlogin, time_ago(mn->lastseen));
 	}
 
 	/* if noone is logged in to this account, show account's lastseen,
 	 * unless we have a nick and it quit at the same time as the account
```

Comment:
Minor nitpick, but could you add a line to the comment to make it clear that if we `hide_info`, we always act as though no one was logged in to avoid leaking that info?